seed_data/management/commands/authorization_user_exam.py

```diff
@@ -0,0 +1,99 @@
+"""
+Authorize for exam of passed users
+"""
+import logging
+import datetime
+from django.core.management import BaseCommand
+
+from courses.models import Program
+from dashboard.api_edx_cache import CachedEdxUserData
+from dashboard.models import CachedEnrollment, ProgramEnrollment
+from dashboard.utils import MMTrack
+from exams.models import (
+    ExamProfile,
+    ExamAuthorization
+)
+from seed_data.utils import add_year
+
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Authorizations of exam of passed users
+    """
+    help = "Trigger exam authorizations when users already passed course"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--username",
+            dest="username",
+            default=None,
+            help="Username who will be authorized for exam if he passed a course(s).",
+            type=str
+        )
+        parser.add_argument(
+            "--program-id",
+            dest="program_id",
+            default=None,
+            help="Program id where we want to authorize users who passed course(s).",
+            type=int
+        )
+
+    def handle(self, *args, **options):
+        """
+        Trigger exam authorizations for user for a course if he has passed it.
+        """
+        program_id = options.get("program_id", None)
+        username = options.get("username", None)
+
+        programs = (
+            Program.objects.filter(id=program_id)
+            if program_id else Program.objects.filter(exam_series_code__isnull=False, live=True)
```

Comment:
@amir-qayyum-khan sorry I didn't specify this before (I thought I had), but yes this is only valid courses where `Program.exam_series_code` and `Course.exam_module` are both set.

I'm also don't think we need this command anymore since we're planning on doing this retroactively for users by marking as paid via a coupon.

The best way forward with this code is probably what you mentioned in chat yesterday which is to create a task that checks all these conditions and then creates the `ExamProfile` and `ExamAuthorization` if they don't already exist and to call that task when they enroll and when we calculate the final grade.

Does that make sense? I want to be sure you're not wasting your time in the wrong direction so if it's easier to discuss in chat please ping me.